brian2/codegen/runtime/cython_rt/templates/common.pyx

```diff
@@ -1,7 +1,8 @@
 #cython: boundscheck=False
 #cython: wraparound=False
-#cython: cdivision=True
+#cython: cdivision=False
```

Comment:
Perhaps we could have a Brian preference that controls whether we do this? Actually, maybe the same preference could control how much care we take to get exactly the same semantics for mod and floordiv too? If the user promised to only use positive values things could be much simpler.

Reply:
I'm not very fond of this, I think this is the kind of detail that users should normally not get into and I'd prefer if model semantics would not depend on a preference (imagine setting it in a configuration file and getting different results depending on where you run a simulation!).

Reply:
Right. I guess I'm fine with it as long as the default is "slow but correct". However, this will complicate the Cython code a bit, since with `cdivision=True` we cannot use `from __future__ import division` and have floating point division for integer values `a/b`.